semantic-ast/src/Main.hs

```diff
@@ -38,16 +43,19 @@ main :: IO ()
 main = generateAST =<< execParser opts
 
 generateAST :: SemanticAST -> IO ()
-generateAST (SemanticAST format source) = do
-  bytestring <- case source of
-    Left filePath -> do
-      Data.ByteString.readFile filePath
+generateAST (SemanticAST format color source) = do
+  bytestrings <- case source of
+    Left filePaths -> do
+      traverse Data.ByteString.readFile filePaths
     Right source -> do
-      pure $ Data.ByteString.Char8.pack source
-  ast <- parseByteString @TreeSitter.Python.AST.Module @(Range, Span) tree_sitter_python bytestring
+      pure [Data.ByteString.Char8.pack source]
```

Comment:
Because I'm binding `source` to a variable, `bytestrings`, the type of that has to be consistent. Therefore, even though `filePaths` is a list, and the source string is _not_ a list, I still have to package it up in a list, or "eliminate" it into a list.